Woof/Woof/Model/Preferences/PreferencesHandler.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+/**
+ A type used for managing user preferences in an application.
+ */
+enum PreferencesHandler {
+    /**
+     Saves the selected role as a user preference.
+
+     - Parameters:
+        - selectedRole: The selected role to be saved.
+
+     - Returns: A boolean value indicating whether the saving operation was successful.
+     */
+    @discardableResult static func saveSelectedRole(_ selectedRole: Role) -> Bool {
+        var preferences: Preferences = loadPreferencesFromStorage() ?? Preferences()
+
+        preferences.selectedRole = selectedRole
+
+        return savePreferencesToStorage(preferences)
+    }
+
+    /**
+     Loads the selected role from user preferences.
+
+     - Returns: The selected role, or `.none` if no preferences are found.
+     */
```

Comment:
The documentation is a little beu confusing and not clear to me.
If the user hasn't selected the role, it also returns `.none`.
```suggestion

     - Returns: The selected role for the user in the app session. If the role couldn't be loaded, returns `.none`.
     */
```

Reply:
you are right your version looks more understandable